test/components/addressbook/addressbook.controller.js

```diff
@@ -69,13 +85,42 @@ describe('AddressbookController', function () {
       beforeEach(function () {
         storageServiceMock.get = sinon.stub().returns(null)
       })
-
       it('sets controller contacts to empty array on invalid return value', () => {
         ctrl.getContacts()
 
         expect(storageServiceMock.get.calledOnce).to.be.true
         expect(ctrl.contacts).to.deep.equal([])
       })
     })
+
+    context("set up addressbook contact modal", () => {
+      it('sets up address book modal', () => {
+        ctrl.addAddressbookContact()
+        sinon.assert.calledOnce(mdDialogShowStub)
+        expect($scope.addAddressbookContact).to.have.all.keys(['add', 'cancel'])
+        expect(typeof $scope.addAddressbookContact.add).to.equal('function')
+        expect(typeof $scope.addAddressbookContact.cancel).to.equal('function')
+      })
+    })
+
+    context("Adding contacts", () => {
```

Comment:
Context can help explain the environment for the asserts within. In your example, there could be asserts for each different path the code could take inside the method. Eg: 
```
context('Adding a contact with an invalid name', () => {
  it('should show an error message', () => {})
})
...
context('Adding a contact with a valid name and address', () => {
  it('should ...', () => {})
  it('should ...', () => {})
})
```

Breaking it up like this makes it a lot easier to see what is being tested.